### ISSUE TITLE
Method mocked crash as not mocked

### ISSUE THREAD

**Issue by jaumard** [nhaarman/mockito-kotlin#211]
I have a weird situation, here is my test: 

```
class HostSelectionInterceptorTest {
    @Rule
    @JvmField
    val mockitoRule: MockitoRule = MockitoJUnit.rule().strictness(Strictness.STRICT_STUBS)

    @Mock private lateinit var baseUrlResolver: BaseUrlResolver
    private lateinit var hostSelectionInterceptor: HostSelectionInterceptor

    @Before
    fun setUp() {
        hostSelectionInterceptor = HostSelectionInterceptor(baseUrlResolver)
    }

    @Test
    fun intercept() {
        val chain = Mockito.mock(Interceptor.Chain::class.java)
        .... //more stubbing
        whenever(baseUrlResolver.getBaseUrl()).thenReturn(Single.just("http://myurl.com"))

        hostSelectionInterceptor.intercept(chain)
        assertEquals("https", hostSelectionInterceptor.scheme)
        assertEquals(443, hostSelectionInterceptor.port)
        assertEquals("myurlmocked.com", hostSelectionInterceptor.host)
    }
}
```

When I run this it crash with NullPointerException (Rx crash because baseUrlResolver.getBaseUrl return null as it's not mocked)
I change MockitoConfiguration to have better error message when something is not mocked and it say `RuntimeException: No mock defined for invocation baseUrlResolver.getBaseUrl();`

If I change the whenever by this:
```
baseUrlResolver = mock {
            on { getBaseUrl() }.doReturn(Single.just("https://myurlmocked.com"))
        }
```
Then it worked but by doing that I'll be able to test only once case when with whenever I can override multiple time what I want to return.

Here is my baseUrlResolver class if it can help:
```
class BaseUrlResolver @Inject constructor(private val mdnsFinder: MdnsFinder, private val preferencesProvider: PreferencesProvider) {
    fun getBaseUrl(): Single<String> {
        return mdnsFinder.searchLISAService()
                .flatMap({ hostPort -> Single.just("http:/" + hostPort) })
                .onErrorResumeNext({ thowable ->
                    val externalUrl: String? = preferencesProvider.getPreferences().get(Preferences.KEY_EXTERNAL_URL)
                    if (externalUrl == null) {
                        return@onErrorResumeNext mdnsFinder.searchLISAService()
                                .flatMap({ hostPort ->
                                    Single.just("http:/" + hostPort)
                                            .onErrorResumeNext { Single.error(thowable) }
                                })
                    } else {
                        return@onErrorResumeNext Single.just(externalUrl)
                    }
                })
    }
}
```

**Comment by nhaarman** [comment]
You have a lot of noise in your example code. Could you create an [MVCE](https://stackoverflow.com/help/mcve)? See https://github.com/nhaarman/mockito-kotlin/issues/207 for example.

**Comment by jaumard** [comment]
Can you run it here https://github.com/mylisabox/lisa-android/blob/master/network/src/test/java/com/mylisabox/network/interceptors/HostSelectionInterceptorTest.kt#L73 as it's open source ? 

If not I'll try to create a more simple example when I've time :)

**Comment by nhaarman** [comment]
Simple examples are always the way to go, I might not always have time to get third party projects to build and get to know them.

**Comment by jaumard** [comment]
@nhaarman I think https://github.com/nhaarman/mockito-kotlin/issues/244 is related to this but I'm not sure, I put a MVCE inside :) 
Again there some RX involved

**Comment by nhaarman** [comment]
Since #244 is resolved and this can't really be reproduced I'm closing it. Please file another issue if it still occurs with the 2.x version.
